File: check_vocabulary.mjs
import { createClient } from '@supabase/supabase-js'
import fs from 'fs'

const supabaseUrl = 'https://hkljprwxvdoxorhcbvpo.supabase.co'
const supabaseKey = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImhrbGpwcnd4dmRveG9yaGNidnBvIiwicm9sZSI6ImFub24iLCJpYXQiOjE3NjE5NjQ0ODYsImV4cCI6MjA3NzU0MDQ4Nn0.RN7ax6dITsRKMLCH9ptQZqYmsh7slXONVHH1Bn3ihwQ'

const supabase = createClient(supabaseUrl, supabaseKey)

async function checkVocabulary() {
  console.log('Fetching vocabulary data from Supabase...')
  
  const { data, error } = await supabase
    .from('vocabulary')
    .select('*')
  
  if (error) {
    console.error('✗ Error fetching vocabulary:', error)
    return
  }
  
  if (data) {
    console.log(`\n✓ Found ${data.length} vocabulary entries`)
    
    // Search for the "acute" error
    const acuteEntries = data.filter(v => 
      v.word && v.word.toLowerCase().includes('acute')
    )
    
    if (acuteEntries.length > 0) {
      console.log(`\n🔍 Found ${acuteEntries.length} entries with 'acute':`)
      acuteEntries.forEach(entry => {
        console.log(`  - ID: ${entry.id}, Word: ${entry.word}, Arabic: ${entry.arabic_translation || 'N/A'}`)
      })
    }
    
    // Save all vocabulary to file for analysis
    fs.writeFileSync('data/vocabulary_dump.json', JSON.stringify(data, null, 2))
    console.log(`\n✓ Vocabulary data saved to data/vocabulary_dump.json`)
    
    // Check for potential errors
    const errors = []
    data.forEach(entry => {
      if (!entry.word) {
        errors.push(`Missing word: ${JSON.stringify(entry)}`)
      }
      if (!entry.arabic_translation) {
        errors.push(`Missing Arabic translation for: ${entry.word}`)
      }
      if (entry.arabic_translation === 'No') {
        errors.push(`Invalid translation 'No' for: ${entry.word} (ID: ${entry.id})`)
      }
    })
    
    if (errors.length > 0) {
      console.log(`\n⚠ Found ${errors.length} potential errors:`)
      errors.slice(0, 20).forEach(error => {
        console.log(`  - ${error}`)
      })
      
      fs.writeFileSync('data/vocabulary_errors.txt', errors.join('\n'))
      console.log(`\n✓ All errors saved to data/vocabulary_errors.txt`)
    } else {
      console.log('\n✓ No errors found in vocabulary data')
    }
  }
}

checkVocabulary()

File: check_vocabulary.py
import os
import json
from supabase import create_client, Client

# Initialize Supabase client
url = "https://hkljprwxvdoxorhcbvpo.supabase.co"
key = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImhrbGpwcnd4dmRveG9yaGNidnBvIiwicm9sZSI6ImFub24iLCJpYXQiOjE3NjE5NjQ0ODYsImV4cCI6MjA3NzU0MDQ4Nn0.RN7ax6dITsRKMLCH9ptQZqYmsh7slXONVHH1Bn3ihwQ"
supabase: Client = create_client(url, key)

# Fetch all vocabulary to check for errors
print("Fetching vocabulary data from Supabase...")
response = supabase.table('vocabulary').select('*').execute()

if response.data:
    print(f"\n✓ Found {len(response.data)} vocabulary entries")
    
    # Search for the "acute" error
    acute_entries = [v for v in response.data if 'acute' in v.get('word', '').lower()]
    
    if acute_entries:
        print(f"\n Found {len(acute_entries)} entries with 'acute':")
        for entry in acute_entries:
            print(f"  - ID: {entry.get('id')}, Word: {entry.get('word')}, Arabic: {entry.get('arabic_translation', 'N/A')}")
    
    # Save all vocabulary to file for analysis
    with open('data/vocabulary_dump.json', 'w', encoding='utf-8') as f:
        json.dump(response.data, f, indent=2, ensure_ascii=False)
    
    print(f"\n✓ Vocabulary data saved to data/vocabulary_dump.json")
    
    # Check for potential errors (missing translations, invalid data)
    errors = []
    for entry in response.data:
        if not entry.get('word'):
            errors.append(f"Missing word: {entry}")
        if not entry.get('arabic_translation'):
            errors.append(f"Missing Arabic translation for: {entry.get('word')}")
        if entry.get('arabic_translation') == 'No':
            errors.append(f"Invalid translation 'No' for: {entry.get('word')} (ID: {entry.get('id')})")
    
    if errors:
        print(f"\n⚠ Found {len(errors)} potential errors:")
        for error in errors[:20]:  # Show first 20 errors
            print(f"  - {error}")
        
        # Save errors to file
        with open('data/vocabulary_errors.txt', 'w', encoding='utf-8') as f:
            f.write('\n'.join(errors))
        print(f"\n✓ All errors saved to data/vocabulary_errors.txt")
    else:
        print("\n✓ No errors found in vocabulary data")
else:
    print("✗ No vocabulary data found")
